lib/importproject.cpp

```diff
@@ -217,13 +217,13 @@ ImportProject::Type ImportProject::import(const std::string &filename, Settings
     return ImportProject::Type::UNKNOWN;
 }
 
-static std::string readUntil(const std::string &command, std::string::size_type *pos, const char until[])
+static std::string readUntil(const std::string& command, std::string::size_type* pos, const char until[], bool skipBackSlash = true)
 {
     std::string ret;
     bool str = false;
     while (*pos < command.size() && (str || !std::strchr(until, command[*pos]))) {
-        if (command[*pos] == '\\')
-            ++*pos;
+        if (skipBackSlash && command[*pos] == '\\')
```

Comment:
hmm.. somehow we want to skip escape backslashes but we do not want to skip path separators. I wonder what would happen if you check if there comes a `"` after then backslashes and if so removes them. It seems strange to parse different compiler options differently.

Reply:
But because the are different we need to parse them differently. Checking for ` " ` will not work because the includes are not necessarily wrapped by ` " `  you see it in the added unit-test

Reply:
sounds strange to me .. I'll need to understand how they are different.. can you show a -D and -I flag with similar information to show how they are different? For instance it would be interesting to see a comparison of `-I"c:\some path"` and `-DFOO="c:\some path"`

Reply:
@danmar that is a good example you already provided.
we first parse for the define e.g. ` -D ` then parse until we hit `=` or space and the we parse again the value of the define, if there is one.
And the first part is different to an include you do not have a `-DC:\\some path`, the path is only in the value.
So parsing the value of a define and parsing an include can be the same